src/Symfony/Component/Messenger/Bridge/MongoDB/Transport/Connection.php

```diff
@@ -0,0 +1,281 @@
+     */
+    public function reject(string $id): bool
+    {
+        try {
+            $deleteResult = $this->collection->deleteOne(['_id' => new ObjectId($id)], $this->getMongoOptions());
+        } catch (\Throwable $exception) {
+            throw new TransportException($exception->getMessage(), 0, $exception);
+        }
+
+        return $deleteResult->getDeletedCount() > 0;
+    }
+
+    public function getMessageCount(): int
+    {
+        return $this->collection->count(
+            $this->createAvailableMessagesQuery()
+        );
+    }
+
+    /**
+     * @throws DriverException
+     */
+    public function find(string $id): ?BSONDocument
+    {
+        return $this->collection->findOne(['_id' => new ObjectId($id)], $this->setTypeMapOption());
+    }
+
+    /**
+     * @return Cursor<BSONDocument>
+     */
+    public function findAll(?int $limit = null): Cursor
+    {
+        $options = [];
+        if (null !== $limit) {
+            $options['limit'] = $limit;
+        }
+
+        return $this->findBy($this->createAvailableMessagesQuery(), $options);
+    }
+
+    /**
+     * @param array<string, mixed>|object $filters
+     * @param array<string, mixed>        $options
+     *
+     * @return Cursor<BSONDocument>
+     */
+    public function findBy($filters, array $options): Cursor
+    {
+        return $this->collection->find($filters, $this->setTypeMapOption($options));
+    }
+
+    /**
+     * @param array<string, mixed>|object $filters
+     * @param array<string, mixed>        $options
+     */
+    public function countBy($filters, array $options): int
+    {
+        return $this->collection->count($filters, $options);
+    }
+
+    public function deleteAll(): void
+    {
+        $this->collection->deleteMany(['queueName' => $this->queueName], []);
+    }
+
+    /**
+     * This method will create the collection and add a compound index
+     * including the queueName, availableAt and deliveredAt fields.
+     */
+    public function setup(): void
+    {
+        $this->collection->createIndex([
+            'availableAt' => 1,
+            'queueName' => 1,
+            'deliveredAt' => 1,
+        ], [
+            'name' => 'facile-it_messenger_index',
+        ]);
+    }
+
+    /**
+     * @return array<string, mixed>
+     */
+    private function createAvailableMessagesQuery(): array
+    {
+        $redeliverLimit = $this->clock->now()->modify(\sprintf('-%d seconds', $this->redeliverTimeout));
+
+        return [
+            '$or' => [
+                ['deliveredAt' => null],
+                ['deliveredAt' => [
+                    '$lt' => new UTCDateTime($redeliverLimit),
+                ]],
+            ],
+            'availableAt' => ['$lte' => new UTCDateTime($this->clock->now())],
+            'queueName' => $this->queueName,
+        ];
+    }
+
+    /**
+     * @return array<string, mixed>
+     */
+    private function getMongoOptions(?Session $session = null): array
```

Comment:
You can be more specific with this method name.
```suggestion
    private function getWriteOptions(?Session $session = null): array
```